src/services/factories/tokenList.ts

```diff
@@ -191,7 +206,7 @@ export function getTokensFactory(factoryParams: {
           `[tokenListFactory][${networkId}] Got details for address ${partialToken.address}: symbol '${partialToken.symbol}' name '${partialToken.name}'`,
         )
         // Get id from address/id mapping
-        const id = addressesAndIds.get(partialToken.address) as number
+        const id = filteredAddressesAndIds.get(partialToken.address) as number
```

Comment:
can't this be `undefined`?

Reply:
I don't think so. The logic for loading the tokens is a bit unnecessarily complex. This is why these things are required.
I did one small refactor in this PR, but they could have more refactors and probably this `as number` would disapear. But this is not part of this PR refactor all.

It cannot happen to be undefined, cause we made sure in the code above that only tokens listed in the exchange (with id) are in this `partialTokens` array